File: presentation/src/main/java/com/nextcloud/talk/models/json/capabilities/ThemingCapability.java
/*
 * Nextcloud Talk application
 *
 * @author Mario Danic
 * Copyright (C) 2017-2019 Mario Danic <mario@lovelyhq.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.nextcloud.talk.models.json.capabilities;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;
import lombok.Data;
import org.parceler.Parcel;

@Parcel
@Data
@JsonObject
class ThemingCapability {
    @JsonField(name = "name")
    String name;

    @JsonField(name = "url")
    String url;

    @JsonField(name = "slogan")
    String slogan;

    @JsonField(name = "color")
    String color;

    @JsonField(name = "color-text")
    String colorText;

    @JsonField(name = "color-element")
    String colorElement;

    @JsonField(name = "logo")
    String logo;

    @JsonField(name = "background")
    String background;

    @JsonField(name = "background-plain")
    boolean backgroundPlain;

    @JsonField(name = "background-default")
    boolean backgroundDefault;
}

File: app/src/main/java/com/nextcloud/talk/models/json/capabilities/ThemingCapability.kt
/*
 * Nextcloud Talk - Android Client
 *
 * SPDX-FileCopyrightText: 2022 Tim Krüger <t@timkrueger.me>
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
package com.nextcloud.talk.models.json.capabilities

import android.os.Parcelable
import com.bluelinelabs.logansquare.annotation.JsonField
import com.bluelinelabs.logansquare.annotation.JsonObject
import kotlinx.parcelize.Parcelize
import kotlinx.serialization.Serializable

@Parcelize
@JsonObject
@Serializable
data class ThemingCapability(
    @JsonField(name = ["name"])
    var name: String?,
    @JsonField(name = ["url"])
    var url: String?,
    @JsonField(name = ["slogan"])
    var slogan: String?,
    @JsonField(name = ["color"])
    var color: String?,
    @JsonField(name = ["color-text"])
    var colorText: String?,
    @JsonField(name = ["color-element"])
    var colorElement: String?,
    @JsonField(name = ["color-element-bright"])
    var colorElementBright: String?,
    @JsonField(name = ["color-element-dark"])
    var colorElementDark: String?,
    @JsonField(name = ["logo"])
    var logo: String?,
    @JsonField(name = ["background"])
    var background: String?,
    @JsonField(name = ["background-plain"])
    var backgroundPlain: Boolean?,
    @JsonField(name = ["background-default"])
    var backgroundDefault: Boolean?
) : Parcelable {
    // This constructor is added to work with the 'com.bluelinelabs.logansquare.annotation.JsonObject'
    constructor() : this(null, null, null, null, null, null, null, null, null, null, null, null)
}
